### PR TITLE
Change simple_table to use 64-bit int and float instead of 32-bit

### PR DESCRIPTION
**Pull request by taldcroft**
The table helper function `simple_table()` was returning a table with 32-bit int and 32-bit float columns.  Now it returns 64-bit values which are more "normal" in everyday use.

Long version:
While there isn't anything specifically wrong with that, it turns out that `np.searchsorted` is quite slow for something like `np.searchsorted(np.arange(1e7, dtype=np.float32), 100.0)` because it casts the input array to the type of the search value (https://github.com/numpy/numpy/issues/5370).  I lost some time trying to figure out why table indexing and np.searchsorted were so slow when in fact it was just that numpy issue.  In order to prevent future digressions because 32-bit values are not the norm, I want to change `simple_table` (which I use all the time in quick testing and development).

### CODE REVIEW

**Comment by mhvk** [comment]
Good idea.  Note the multiple test failures. Also, see in-line comment.

File: astropy/table/table_helpers.py [hunk 1]
```diff
@@ -96,9 +96,9 @@ def simple_table(size=3, cols=None, kinds='ifS', masked=False):
     letters = np.array([c for c in string.ascii_letters])
     for jj, kind in zip(range(cols), cycle(kinds)):
         if kind == 'i':
-            data = np.arange(1, size + 1, dtype=int) + jj
+            data = np.arange(1, size + 1, dtype=np.int64) + jj
```
Comment (mhvk) [inline astropy/table/table_helpers.py]: More out of curiosity than anything else: Is this really the underlying problem?  Or is it just the `dtype=kind` line? On my machine at least:
```
In [2]: np.arange(10, dtype=int).dtype
Out[2]: dtype('int64')
In [3]: Column(np.arange(10, dtype=int), dtype='i').dtype
Out[3]: dtype('int32')
```
Or is this different on a 32-bit machine? Anyway, explicit is better than implicit, so no reason to change!

Comment (taldcroft) [inline astropy/table/table_helpers.py]: I was mostly worried about Windows where int32 is the default, so I'm not sure what `dtype=int` does, but I'm guessing it's int32.

Comment (embray) [inline astropy/table/table_helpers.py]: Yup--this will ensure more consistent results.

**Comment by taldcroft** [comment]
Passing tests now.

**Comment by mhvk** [comment]
OK, looks good; merging.

**Pull request merged**